velox/common/memory/MmapAllocator.cpp

```diff
@@ -657,50 +657,57 @@ bool MmapAllocator::SizeClass::allocate(
 }
 
 bool MmapAllocator::SizeClass::allocateLocked(
-    const ClassPageCount numPages,
+    ClassPageCount numClassPages,
     MachinePageCount* numUnmapped,
     Allocation& out) {
   const size_t numWords = pageBitmapSize_;
-  ClassPageCount considerMappedOnly = std::min(numMappedFreePages_, numPages);
-  auto numPagesToAllocate = numPages;
-  if (considerMappedOnly > 0) {
-    const auto previousPages = out.numPages();
-    allocateFromMappedFree(considerMappedOnly, out);
-    const auto numAllocated = (out.numPages() - previousPages) / unitSize_;
-    VELOX_CHECK_EQ(
-        numAllocated,
-        considerMappedOnly,
-        "Allocated different number of pages");
-    numMappedFreePages_ -= numAllocated;
-    numPagesToAllocate -= numAllocated;
-  }
-  if (numPagesToAllocate == 0) {
-    return true;
-  }
-  if (numUnmapped == nullptr) {
-    return false;
-  }
-  uint32_t cursor = clockHand_;
-  int32_t numWordsTried = 0;
-  for (;;) {
-    auto previousCursor = cursor;
-    if (++cursor >= numWords) {
-      cursor = 0;
+  const auto maxClassPagesInRun = static_cast<ClassPageCount>(
+      Allocation::PageRun::kMaxPagesInRun / unitSize_);
+  while (numClassPages > 0) {
+    auto numClassPagesToAllocate = std::min(numClassPages, maxClassPagesInRun);
```

Comment:
```
const auto numClassPagesToAllocate =
```

Reply:
We update this variable below inside `if (considerMappedOnly > 0) {`